experimental/zerotrie/src/cursor.rs

```diff
@@ -87,6 +135,15 @@ pub struct ZeroTrieSimpleAsciiCursor<'a> {
     trie: ZeroTrieSimpleAscii<&'a [u8]>,
 }
 
+/// A cursor into a [`ZeroAsciiIgnoreCaseTrie`], useful for stepwise lookup.
+///
+/// For examples, see [`ZeroAsciiIgnoreCaseTrie::cursor()`].
+// Clone but not Copy: <https://stackoverflow.com/q/32324251/1407170>
+#[derive(Debug, Clone)]
+pub struct ZeroAsciiIgnoreCaseTrieCursor<'a> {
```

Comment:
not a big fan of adding yet another type, but you already know that

Reply:
I wrote some ideas on how to resolve this in https://github.com/unicode-org/icu4x/issues/4561